src/SessionManager.cc

```diff
@@ -425,17 +410,20 @@ void SessionManager::Clear()
 
 void SessionManager::GetStats(SessionStats& s)
 	{
-	s.max_TCP_conns = stats.GetOrAdd({{"tcp", "max_conns"}}).Value();
-	s.num_TCP_conns = stats.GetOrAdd({{"tcp", "num_conns"}}).Value();
-	s.cumulative_TCP_conns = stats.GetOrAdd({{"tcp", "cumulative_conns"}}).Value();
+	auto* tcp_stats = stats.GetCounters("tcp");
```

Comment:
It's technically cached in the manager already in the `StatsBlock` object. The call to `GetCounter()` in `GetStats()` ends up just being a map lookup in normal cases. 

I could see it being slightly slower if you only got one protocol's data. That would cause it to make a couple of extra calls to `GetOrAdd()` for the others, which would then only return zero values. How frequently is something like `get_conn_stats()` (which calls `GetStats()`), and would those extra calls be significant?

I was already planning to rework this code a bit anyways to loop through the map of blocks and only return data for the existing ones, since that would also make it return stats for non-tcp/udp/icmp data, but not during this PR.